src/test/java/uk/ac/ebi/atlas/experiments/ExperimentCollectionDaoIT.java

```diff
@@ -0,0 +1,51 @@
+package uk.ac.ebi.atlas.experiments;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.transaction.annotation.Transactional;
+import uk.ac.ebi.atlas.configuration.TestConfig;
+
+import javax.inject.Inject;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@ExtendWith(SpringExtension.class)
+@WebAppConfiguration
+@ContextConfiguration(classes = TestConfig.class)
+@Transactional
+@Sql("/fixtures/experiment-fixture.sql")
+@Sql("/fixtures/scxa-collections-fixture.sql")
+@Sql("/fixtures/scxa-experiment2collection-fixture.sql")
+class ExperimentCollectionDaoIT {
+
+    @Inject
+    private JdbcTemplate jdbcTemplate;
+
+    @Inject
+    ExperimentCollectionDao subject;
+
+    @BeforeEach
+    void setUp() {
+        subject = new ExperimentCollectionDao(jdbcTemplate);
+    }
+
+    @Test
+    void ifNoExperimentCollectionReturnNothing() {
+        assertThat(subject.getExperimentCollections("E-FOO-0000"))
+                .isEmpty();
+    }
+
+    @Test
+    void ifExperimentCollectionExistsReturnCollectionName() {
+        assertThat(subject.getExperimentCollections("E-EHCA-2"))
```

Comment:
If all our experiments are mapped to a collection I’d have an assertion in this test that picks one at random and check that the value is not empty. Just to have something a bit more general.

Reply:
I get your point. I am not changing the here at the moment because for now such test structure will pass in my local db but might fail on bamboo because atm not all experiments are mapped to a collection so if I pick up a one experiment accession at random then it might not have collection mapped to it resulting in failed test.